src/mp-instance.ts

```diff
@@ -126,11 +126,12 @@ export default function mParticleInstance(this: IMParticleWebSDKInstance, instan
     this._IntegrationCapture = new IntegrationCapture();
 
     // required for forwarders once they reference the mparticle instance
-    this.IdentityType = IdentityType;
-    this.EventType = EventType as unknown as valueof<typeof EventType>;
+    this.IdentityType = IdentityType as unknown as valueof<typeof IdentityType>;
+    // this.EventType = EventType as unknown as valueof<typeof EventType>;
+    this.EventType = EventType;
     this.CommerceEventType = CommerceEventType as unknown as valueof<typeof CommerceEventType>;
     this.PromotionType = PromotionActionType as unknown as valueof<typeof PromotionActionType>;
-    this.ProductActionType = ProductActionType as unknown as valueof<typeof ProductActionType>;
+    this.ProductActionType = ProductActionType;
```

Comment:
similarly - did we not need the `as unknown as valueof<...>` for all of these after all?

Reply:
I noticed that when actually using the assigned values with `unknown as valueof`, it was treating them as `any`. This way, in the tests, we can validate the actual values.